pkg/rollout/rollout.go

```diff
@@ -200,6 +208,41 @@ func (r *Rollout) PrepareRollback(svc *run.Service, stable, candidate string) *r
 	return svc
 }
 
+// updateServiceBasedOnDiagnosis updates a service's traffic configuration
+// based on the diagnosis.
+// If service was unchanged, nil is returned.
+//
+// TODO (gvso): Refactor this and other methods to return only a traffic object.
+// Then, rename this to trafficBasedOnDiagnosis.
+// Also, move those traffic-config-related methods to traffic.go file.
+func (r *Rollout) updateServiceBasedOnDiagnosis(svc *run.Service, diagnosis health.DiagnosisResult, stable, candidate string) (*run.Service, error) {
+	switch diagnosis {
+	case health.Inconclusive:
+		r.log.Debug("health check inconclusive")
+		return nil, nil
+	case health.Healthy:
+		r.log.Debug("healthy candidate")
+		enoughTime, err := r.shouldRollForward(svc.Metadata.Annotations[LastRollForwardAnnotation], r.strategy.TimeBetweenRollForwardsMins)
+		if err != nil {
+			return nil, errors.Wrap(err, "could not determine if roll forward is allowed")
+		}
+		if !enoughTime {
+			r.log.Debug("no enough time since last roll forward, service unchanged")
+			return nil, nil
```

Comment:
this should be inconclusive but we need to figure out how to return a human-readable explanation for the status annotation